pkg/adapter/nimbus-kyverno/processor/kpbuilder.go

```diff
@@ -239,75 +351,313 @@ func cocoRuntimeAddition(np *v1alpha1.NimbusPolicy) ([]kyvernov1.Policy, error)
+				Name: generatorPolicyName,
+			},
+			Spec: kyvernov1.Spec{
+				GenerateExisting: true,
+				Rules: []kyvernov1.Rule{
+					{
+						Name: cve + "virtual-patch-karmor",
+						MatchResources: kyvernov1.MatchResources{
+							Any: kyvernov1.ResourceFilters{
+								{
+									ResourceDescription: kyvernov1.ResourceDescription{
+										Kinds: []string{
+											"v1/Pod",
+										},
+										Selector: &metav1.LabelSelector{
+											MatchLabels: labels,
+										},
+									},
+								},
+							},
+						},
+						RawAnyAllConditions: &v1.JSON{Raw: preconditionBytes},
+						Context: []kyvernov1.ContextEntry{
+							{
+								Name: "containerName",
+								Variable: &kyvernov1.Variable{
+									JMESPath: jmesPathContainerNameQuery,
+								},
+							},
+							getPodName,
+						},
+						Generation: kyvernov1.Generation{
+							ResourceSpec: kyvernov1.ResourceSpec{
+								APIVersion: "security.kubearmor.com/v1",
+								Kind:       "KubeArmorPolicy",
+								Name:       generatedPolicyName,
+								Namespace:  np.GetNamespace(),
+							},
+							RawData: &v1.JSON{Raw: policyBytes},
 						},
 					},
 				},
-			}
-			matchResourceFilters = append(matchResourceFilters, resourceFilter)
+			},
 		}
-	} else {
-		resourceFilter := kyvernov1.ResourceFilter{
-			ResourceDescription: kyvernov1.ResourceDescription{
-				Kinds: []string{
-					"v1/Pod",
+	}
+
+	// kyverno policy generation
+
+	if polengine == "kyverno" {
+
+		generatedPolicyName := metadataMap["name"].(string)
+		selectorMap := map[string]any{
+			"matchLabels": labels,
+		}
+
+		kindMap := map[string]any{
+			"kinds": []any{
+				"Pod",
+			},
+			"selector": selectorMap,
+		}
+
+		newMatchMap := map[string]any{
+			"any": []any{
+				map[string]any{
+					"resources": kindMap,
 				},
 			},
 		}
-		matchResourceFilters = append(matchResourceFilters, resourceFilter)
-	}
+		rulesMap := specMap["rules"].([]any)
+		rule := rulesMap[0].(map[string]any)
 
-	background := true
-	kp := kyvernov1.Policy{
-		Spec: kyvernov1.Spec{
-			Background: &background,
-			Rules: []kyvernov1.Rule{
-				{
-					Name: "pod-security-standard",
-					MatchResources: kyvernov1.MatchResources{
-						Any: matchResourceFilters,
-					},
-					Validation: kyvernov1.Validation{
-						PodSecurity: &kyvernov1.PodSecurity{
-							Level:   psa_level,
-							Version: "latest",
+		// adding resources as Pod and ommitting all the incoming resource types
+		delete(rule, "match")
+		rule["match"] = newMatchMap
+
+		// appending the image matching precondition to the existing preconditions
+		preCndMap := rule["preconditions"].(map[string]any)
+		conditionsList, ok := preCndMap["any"].([]any)
+		if ok {
+			preConditionMap["all"] = append(preConditionMap["all"].([]any), conditionsList...)
+		}
+
+		delete(rule, "preconditions")
+
+		rule["preconditions"] = preConditionMap
+
+		policyBytes, err := json.Marshal(policyData)
+		if err != nil {
+			panic(err.Error())
```

Comment:
Please do not use `panic` handle the error gracefully instead.
Could you explain why are you using `panic` here?